modules/apps/foundation/address-formatter/address-formatter-api/src/main/java/com/liferay/address/formatter/util/AddressFormatterUtil.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.address.formatter.util;
+
+import com.liferay.address.formatter.AddressFormatter;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMap;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMapFactory;
+import com.liferay.portal.kernel.model.Address;
+import com.liferay.portal.kernel.model.Country;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * @author Pei-Jung Lan
+ */
+@Component(immediate = true, service = AddressFormatterUtil.class)
+public class AddressFormatterUtil {
+
+	public static String format(Address address) {
+		Country country = address.getCountry();
+
+		String a2 = country.getA2();
+
+		AddressFormatter addressFormatter = getAddressFormatter(a2);
+
+		return addressFormatter.format(address);
+	}
+
+	public static AddressFormatter getAddressFormatter(String countryA2) {
```

Comment:
Hey Pei-Jung,

Thanks for the detailed response. Everything you say makes sense.

Reading the story I think that it's not entirely clear, so what I'd like to ask you to do is to update it to ensure it describes well what we are implementing. Then you can check with Sam Kong (and the other people involved in the definition) to double check that it meets their needs.

Regarding Formatters overlapping I'm referring to a case like a JapanFormatter detecting the address is Romanized and then having to choose whether to also know how to format that case or else having to invoke another formatter (which is not guaranteed to be available).

What do you think we should do in this case?